bigquery/tests/unit/test_client.py

```diff
@@ -1546,6 +1546,242 @@ def test_query_rows_w_job_config(self):
+        http = object()
+        client = self._make_one(project=PROJECT, credentials=creds, _http=http)
+        conn = client._connection = _Connection(DATA)
+        table_ref = DatasetReference(PROJECT, DS_ID).table(TABLE_ID)
+        color = SchemaField('color', 'STRING', mode='REPEATED')
+        index = SchemaField('index', 'INTEGER', 'REPEATED')
+        score = SchemaField('score', 'FLOAT', 'REPEATED')
+        struct = SchemaField('struct', 'RECORD', mode='REPEATED',
+                             fields=[index, score])
+
+        iterator = client.list_rows(table_ref, selected_fields=[color, struct])
+        page = six.next(iterator.pages)
+        rows = list(page)
+        total_rows = iterator.total_rows
+        page_token = iterator.next_page_token
+
+        self.assertEqual(len(rows), 1)
+        self.assertEqual(rows[0][0], ['red', 'green'])
+        self.assertEqual(rows[0][1], [{'index': [1, 2],
+                                       'score': [3.1415, 1.414]}])
+        self.assertEqual(total_rows, ROWS)
+        self.assertEqual(page_token, TOKEN)
+
+        self.assertEqual(len(conn._requested), 1)
+        req = conn._requested[0]
+        self.assertEqual(req['method'], 'GET')
+        self.assertEqual(req['path'], '/%s' % PATH)
+
+    def test_list_rows_w_record_schema(self):
+        import six
+        from google.cloud.bigquery.dataset import DatasetReference
+        from google.cloud.bigquery.table import Table, SchemaField
+
+        PROJECT = 'PROJECT'
+        DS_ID = 'DS_ID'
+        TABLE_ID = 'TABLE_ID'
+        PATH = 'projects/%s/datasets/%s/tables/%s/data' % (
+            PROJECT, DS_ID, TABLE_ID)
+        ROWS = 1234
+        TOKEN = 'TOKEN'
+        DATA = {
+            'totalRows': ROWS,
+            'pageToken': TOKEN,
+            'rows': [
+                {'f': [
+                    {'v': 'Phred Phlyntstone'},
+                    {'v': {'f': [{'v': '800'}, {'v': '555-1212'}, {'v': 1}]}},
+                ]},
+                {'f': [
+                    {'v': 'Bharney Rhubble'},
+                    {'v': {'f': [{'v': '877'}, {'v': '768-5309'}, {'v': 2}]}},
+                ]},
+                {'f': [
+                    {'v': 'Wylma Phlyntstone'},
+                    {'v': None},
+                ]},
+            ]
+        }
+        creds = _make_credentials()
+        http = object()
+        client = self._make_one(project=PROJECT, credentials=creds, _http=http)
+        conn = client._connection = _Connection(DATA)
+        table_ref = DatasetReference(PROJECT, DS_ID).table(TABLE_ID)
+        full_name = SchemaField('full_name', 'STRING', mode='REQUIRED')
+        area_code = SchemaField('area_code', 'STRING', 'REQUIRED')
+        local_number = SchemaField('local_number', 'STRING', 'REQUIRED')
+        rank = SchemaField('rank', 'INTEGER', 'REQUIRED')
+        phone = SchemaField('phone', 'RECORD', mode='NULLABLE',
+                            fields=[area_code, local_number, rank])
```

Comment:
I'd love to see a repeated field inside a record field.

Reply:
That is in test_list_rows_repeated_fields. See the definition of `struct`.

Reply:
👍